src/code/z_message_nes.c

```diff
@@ -1,18 +1,271 @@
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x73U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x73;
+
+    f += 16.0f * msgCtx->unk12098 * 6.0f;
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadLocalizedRupeesNES(GlobalContext* globalCtx, s16* decodedBufPos, s32* offset, f32* arg3) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+    u8 j;
+
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x20;
+    p = p + 1;
+    for (j = 0; j < D_801D0710[gSaveContext.language - 1]; j++) {
+        Font_LoadCharNES(globalCtx, D_801D06F0[gSaveContext.language - 1][j], o);
+        ((u8*)msgCtx->decodedBuffer)[p] = D_801D06F0[gSaveContext.language - 1][j];
+        o += FONT_CHAR_TEX_SIZE;
+        p++;
+    }
+    f += 16.0f * msgCtx->unk12098 * (D_801D0710[gSaveContext.language - 1] + 1);
+    p--;
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadRupeesNES(GlobalContext* globalCtx, s16* decodedBufPos, s32* offset, f32* arg3, s16 singular) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x20;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x52U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x52;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x75U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x75;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x70U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x70;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x65;
+    p++;
+    Font_LoadCharNES(globalCtx, 0x65U, o);
+    o += FONT_CHAR_TEX_SIZE;
+    ((u8*)msgCtx->decodedBuffer)[p] = 0x65;
+    if (singular != 1) {
+        p++;
+        Font_LoadCharNES(globalCtx, 0x73U, o);
+        o += FONT_CHAR_TEX_SIZE;
+        ((u8*)msgCtx->decodedBuffer)[p] = 0x73;
+        f += 16.0f * msgCtx->unk12098 * 6.0f;
+    } else {
+        f += 16.0f * msgCtx->unk12098 * 5.0f;
+    }
+    *decodedBufPos = p;
+    *offset = o;
+    *arg3 = f;
+}
+
+void Message_LoadTimeNES(GlobalContext* globalCtx, u8 arg1, s32* offset, f32* arg3, s16* decodedBufPos) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 p = *decodedBufPos;
+    s32 o = *offset;
+    f32 f = *arg3;
+    u32 dayTime;
+    s16 digits[4];
+    f32 timeInSeconds;
+    s32 day;
+    s16 i;
+
+    if (arg1 == 0xCF) {
+        dayTime = 0x40000 - (((day = gSaveContext.day) % 5) << 16) - (u16)(-0x4000 + gSaveContext.time);
```

Comment:
Oh, a nasty time calculation.
I wonder if the CURRENT_DAY macro would match here
```suggestion
        dayTime = 0x40000 - (CURRENT_DAY << 16) - (u16)(-0x4000 + gSaveContext.time);
```

Reply:
I tried and CURRENT_DAY doesn't match, this is a bit better though 
```C
day = gSaveContext.day;
dayTime = 0x40000 - ((day % 5) << 16) - (u16)(-0x4000 + gSaveContext.time);
```

Reply:
Interesting, this exact same construct is also used in `EnBji01`. Wonder if this would be worthy of a macro?

Reply:
Yeah, those two are very similar, but not exactly the same. Making a macro which matches for both doesn't sound very straight forward. It is something to consider tho, specially if this pops up again

Reply:
Just want to note that while I had this checked out, I tested just copy pasting the same code from `EnBji01` and it does work here as well so a macro should work out. 
Something like 
```c
TIME_BEFORE_MOON_CRASH(day, time) = ((-((day % 5) << 0x10) - ((u16)(time - 0x4000))) + 0x40000)
```
However if we decide to wait for more uses that would be fine, though we might want consider marking these so we don't forget about them.